### PR TITLE
Allow searching system-wide data path on Linux for userscripts

### PR DESCRIPTION
**Pull request by craftyguy**
This implements a system-wide data location (/usr/share/qutebrowser) when running on Linux, and the userscript command has been modified to search in this location if it cannot find a script in the user-specific path that the standarddir.data() returns.
For non-Linux systems and Linux systems where nothing was installed to /usr/share/qutebrowser, the system_data() function defaults to calling data() for the user-specific path. This insures that there is always _something_ usable returned by this function, even if the path is not system-wide. 

Packagers would need to create this path (/usr/share/qutebrowser) on Linux systems, and install the userscripts directory to this location. In the future, other files could be placed here (global qutebrowser.conf, anyone? :P)

---
This change is [Reviewable](https://reviewable.io/reviews/the-compiler/qutebrowser/1356)

### CODE REVIEW

File: tests/unit/utils/test_standarddir.py [hunk 1]
```diff
@@ -313,3 +313,32 @@ def test_exists_race_condition(self, mocker, tmpdir, typ):
...
+    def test_system_datadir_exist(self, monkeypatch):
+        """Test that system-wide path isn't used on linux if path not exist."""
+        monkeypatch.setattr('sys.platform', "linux")
+        monkeypatch.setattr(os.path, 'exists', lambda path: False)
+        assert standarddir.system_data() != "/usr/share/qutebrowser"
+
+    def test_system_datadir_notexist(self, monkeypatch):
+        """Test that system-wide path is not used if it doesn't exist."""
+        monkeypatch.setattr(os.path, 'exists', lambda path: False)
+        assert standarddir.system_data() != "/urs/share/qutebrowser"
```
Comment (The-Compiler) [inline tests/unit/utils/test_standarddir.py]: I don't think it's ever going to equal `/urs/...` :wink:

File: tests/unit/utils/test_standarddir.py [hunk 2]
```diff
@@ -313,3 +313,32 @@ def test_exists_race_condition(self, mocker, tmpdir, typ):
...
+
+    @pytest.mark.windows
+    def test_system_datadir_unsupportedos(self, monkeypatch):
+        """Test that system-wide path is not used on non-Linux OS."""
+        assert standarddir.system_data() != "/urs/share/qutebrowser"
```
Comment (The-Compiler) [inline tests/unit/utils/test_standarddir.py]: Same as above

**Comment by The-Compiler** [comment]
Odd, seems like something with `standarddir.data()` is broken on Windows (as it tries to create `C:\`). I'll take a look and see why the heck tests didn't catch this later.

**Comment by craftyguy** [comment]
Weird, since it just falls back to calling data(), which should be returning a valid path and not C:\, right?

**Comment by The-Compiler** [comment]
For posterity: The `data()` call failed because the test patched `sys.platform` to be `linux`, which was a bad idea.

File: tests/unit/utils/test_standarddir.py [hunk 3]
```diff
@@ -313,3 +313,25 @@ def test_exists_race_condition(self, mocker, tmpdir, typ):
...
+    @pytest.mark.linux
+    def test_system_datadir_not_exist_linux(self, monkeypatch):
+        """Test that system-wide path isn't used on linux if path not exist."""
+        monkeypatch.setattr(os.path, 'exists', lambda path: False)
+        assert standarddir.system_data() != "/usr/share/qutebrowser"
+
+    def test_system_datadir_unsupportedos(self, monkeypatch):
+        """Test that system-wide path is not used on non-Linux OS."""
+        monkeypatch.setattr('sys.platform', "potato")
+        assert standarddir.system_data() != "/urs/share/qutebrowser"
```
Comment (The-Compiler) [inline tests/unit/utils/test_standarddir.py]: This one is still `urs` :wink:

Also, any reason you didn't change this to `assert standarddir.system_data() == standarddir.data()` like I proposed?

**Comment by craftyguy** [comment]
No reason other than I must have missed your comment :(
I'll change/fix and amend.

**Comment by craftyguy** [comment]
Test has been changed based on your feedback!

File: qutebrowser/commands/userscripts.py [hunk 4]
```diff
@@ -344,13 +344,17 @@ def run(cmd, *args, win_id, env, verbose=False):
...
-        cmd = os.path.join(standarddir.data(), "userscripts", cmd)
+    if not os.path.isabs(cmd_path):
+        log.misc.debug("{} is no absolute path".format(cmd_path))
+        cmd_path = os.path.join(standarddir.data(), "userscripts", cmd)
+        if not os.path.exists(cmd_path):
+            cmd_path = os.path.join(standarddir.system_data(),
+                                    "userscripts", cmd)
+    log.misc.debug("Userscript to run: {}".format(cmd_path))
 
     runner.run(cmd, *args, env=env, verbose=verbose)
```
Comment (The-Compiler) [inline qutebrowser/commands/userscripts.py]: Shouldn't this be `cmd_path`?

**Comment by craftyguy** [comment]
I'm not sure I follow... or line commenting/highlighting on github isn't displaying properly in qutebrowser :P. What are you referring to?

**Comment by The-Compiler** [comment]
You're setting `cmd_path`, but then never actually using it, and instead running `cmd`.

**Comment by craftyguy** [comment]
Crap, I see, I thought you were referring to something I had modified. Runner.run now calls cmd_path!

**Comment by The-Compiler** [comment]
Yeah, seems to work now :tada: - thank you!

**Pull request merged**